sdk/identity/identity/src/credentials/cliCredential.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from '../util/logging';
+import { CredentialClient, CliCredentialClient, CliCredentialOptions } from '../client/CliCredentialClient';
+
+/**
+ * Provides the user access token and expire time
+ * with azure cli command "az account get-access-token" in powershell.  
+ */
+export class CliCredential implements TokenCredential {
+  private client: CredentialClient;
+  /**
+   * Creates an instance of the CliCredential class.
+   *
+   * @param options Options for configuring the client which makes the authentication request.
+   */
+  constructor(options?: CliCredentialOptions) {
+    if (options && options.cliCredentialClient) {
+      this.client = options.cliCredentialClient;
+    } else {
+      this.client = new CliCredentialClient();
+    }
+  }
+
+  /**
+  * Authenticates with Azure Active Directory and returns an access token if
+  * successful.  If authentication cannot be performed at this time, this method may
+  * return null.  If an error occurs during authentication, an {@link AuthenticationError}
+  * containing failure details will be thrown.
+  *
+  * @param scopes The list of scopes for which the token will have access.
+  * @param options The options used to configure any requests this
+  *                TokenCredential implementation might make.
+  */
+  public async getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null> {
+    const { span } = createSpan("CliCredential-getToken", options);
+
+    return new Promise((resolve, reject) => {
+      let scope: string;
+      scope = typeof scopes === "string" ? scopes : scopes[0];
+      logger.info(`use the scope ${scope}`);
+      const resource = scope.replace(/\/.default$/, "");
+      let responseData = "";
+
+      const { span } = createSpan("CliCredential-getToken", options);
+      this.client.createProcess(`az account get-access-token --resource ${resource}`).then(
```

Comment:
We should pass `--output json` explicitly, so if the user has a different default output configured, we still print JSON.